### PR TITLE
gorilla improved perf by bitwise operations

### PR DESCRIPTION
**Pull request by OfirMos**

### CODE REVIEW

File: src/gorilla.c [hunk 1]
```diff
@@ -218,7 +220,7 @@ static bool Bin_InRange(int64_t x, u_int8_t nbits) {
 }
 
 static inline bool Bins_bitoff(const u_int64_t *bins, globalbit_t bit) {
-    return !(bins[bit / BINW] & BIT(localbit(bit)));
+    return !(bins[bit >> BINW_SHIFT] & BIT(localbit(bit)));
```
Comment (filipecosta90) [inline src/gorilla.c:223]: @OfirMos in the past we've tested it and saw no effect on the change. No arm done but we should prove if this approach is not already done by the compiler during build time...

Comment (OfirMos) [reply src/gorilla.c:223]: I think the change have improved almost all scenarios by 10%, I have now repushed trying to check if changing to macro will improve it further.

Comment (filipecosta90) [reply src/gorilla.c:223]: @OfirMos I've used the latest data from master and this branch and compared both achievable ops/sec and p50 latency. 

# achievable ops/sec (   `15 stable tests between versions.` )
```
$ redisbench-admin compare --github_repo RedisTimeSeries --github_org RedisTimeSeries --triggering_env circleci --baseline-branch master --comparison-branch perf_gorilla_improved_perf_using_bitwise --metric_name Totals.overallQueryRates.all_queries
Effective log level set to INFO
2022-01-03 10:42:28,201 INFO Using: redisbench-admin 0.6.0
2022-01-03 10:42:28,201 INFO Checking connection to RedisTimeSeries with user: None, host: benchmarks.redislabs.com, port: 12000
2022-01-03 10:42:28,782 INFO Using deployment_type=oss-standalone and deployment_name=oss-standalone for the analysis
2022-01-03 10:42:28,912 WARNING Based on test-cases set (key=ci.benchmarks.redislabs/circleci/RedisTimeSeries/RedisTimeSeries:testcases) we have 49 distinct benchmarks. 
2022-01-03 10:42:37,771 INFO Printing differential analysis between branches
# Comparison between master and perf_gorilla_improved_perf_using_bitwise for metric: Totals.overallQueryRates.all_queries
|             Test Case             |Baseline value|Comparison Value|% change (higher-better)|
|-----------------------------------|-------------:|---------------:|-----------------------:|
|tsbs-scale100_cpu-max-all-1        |      1578.758|        1585.134|                    0.40|
|tsbs-scale100_cpu-max-all-8        |       208.187|         208.291|                    0.05|
|tsbs-scale100_double-groupby-1     |       111.300|         113.585|                    2.05|
|tsbs-scale100_double-groupby-5     |        22.868|          22.921|                    0.23|
|tsbs-scale100_double-groupby-all   |        11.705|          11.736|                    0.27|
|tsbs-scale100_groupby-orderby-limit|       953.740|         957.220|                    0.36|
|tsbs-scale100_high-cpu-1           |       251.989|         251.286|                   -0.28|
|tsbs-scale100_high-cpu-all         |         3.990|           3.994|                    0.09|
|tsbs-scale100_lastpoint            |       564.015|         573.041|                    1.60|
|tsbs-scale100_single-groupby-1-1-1 |     21339.220|       20878.140|                   -2.16|
|tsbs-scale100_single-groupby-1-1-12|      4354.202|        4334.084|                   -0.46|
|tsbs-scale100_single-groupby-1-8-1 |      2531.829|        2531.953|                    0.00|
|tsbs-scale100_single-groupby-5-1-1 |      1483.184|        1483.308|                    0.01|
|tsbs-scale100_single-groupby-5-1-12|       452.578|         453.251|                    0.15|
|tsbs-scale100_single-groupby-5-8-1 |       721.572|         721.783|                    0.03|
2022-01-03 10:42:37,810 INFO Detected a total of 15 stable tests between versions.
```
# p50 (   `14 stable tests between versions and 1 regression.` )
```
$ redisbench-admin compare --github_repo RedisTimeSeries --github_org RedisTimeSeries --triggering_env circleci --baseline-branch master --comparison-branch perf_gorilla_improved_perf_using_bitwise --metric_name Totals.overallQuantiles.all_queries.q50 --metric_mode 'lower-better'
Effective log level set to INFO
2022-01-03 10:43:17,822 INFO Using: redisbench-admin 0.6.0
2022-01-03 10:43:17,822 INFO Checking connection to RedisTimeSeries with user: None, host: benchmarks.redislabs.com, port: 12000
2022-01-03 10:43:18,212 INFO Using deployment_type=oss-standalone and deployment_name=oss-standalone for the analysis
2022-01-03 10:43:18,332 WARNING Based on test-cases set (key=ci.benchmarks.redislabs/circleci/RedisTimeSeries/RedisTimeSeries:testcases) we have 49 distinct benchmarks. 
2022-01-03 10:43:27,653 INFO Printing differential analysis between branches
# Comparison between master and perf_gorilla_improved_perf_using_bitwise for metric: Totals.overallQuantiles.all_queries.q50
|             Test Case             |Baseline value|Comparison Value|% change (lower-better)|
|-----------------------------------|-------------:|---------------:|----------------------:|
|tsbs-scale100_cpu-max-all-1        |        40.959|          40.731|                   0.56|
|tsbs-scale100_cpu-max-all-8        |       240.231|         235.895|                   1.84|
|tsbs-scale100_double-groupby-1     |       605.119|         586.783|                   3.12|
|tsbs-scale100_double-groupby-5     |      2964.607|        2959.743|                   0.16|
|tsbs-scale100_double-groupby-all   |      5894.911|        5703.167|                   3.36|
|tsbs-scale100_groupby-orderby-limit|        59.975|          59.565|                   0.69|
|tsbs-scale100_high-cpu-1           |       162.975|         177.767|                  -8.32|
|tsbs-scale100_high-cpu-all         |     16095.231|       16315.391|                  -1.35|
|tsbs-scale100_lastpoint            |       111.303|         109.563|                   1.59|
|tsbs-scale100_single-groupby-1-1-1 |         3.155|           3.005|                   4.99|
|tsbs-scale100_single-groupby-1-1-12|        14.626|          14.594|                   0.22|
|tsbs-scale100_single-groupby-1-8-1 |        25.440|          25.429|                   0.04|
|tsbs-scale100_single-groupby-5-1-1 |        43.339|          43.363|                  -0.06|
|tsbs-scale100_single-groupby-5-1-12|       138.927|         138.679|                   0.18|
|tsbs-scale100_single-groupby-5-8-1 |        88.611|          88.551|                   0.07|
2022-01-03 10:43:27,729 INFO Detected a total of 14 stable tests between versions.
2022-01-03 10:43:27,729 WARNING Detected a total of 1 regressions bellow the regression water line (< -5.0%)
2022-01-03 10:43:27,729 WARNING Printing BENCHMARK env var compatible list
2022-01-03 10:43:27,729 WARNING BENCHMARK=tsbs-scale100_high-cpu-1.yml
```

Are you sure about the positive impact?

Comment (filipecosta90) [reply src/gorilla.c:223]: just saw the "closed" update. all good

**Pull request closed without merging**